### PR TITLE
Th2 1819 Added converter JavaType to class in ProtoToIMessageConverter

### PR DESCRIPTION
**Pull request by Max-81**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -55,6 +56,11 @@ dependencies {
 
     // Use JUnit Jupiter Engine for testing.
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.1'
+
+    implementation 'org.openjdk.jmh:jmh-core:1.28'
+    implementation 'org.openjdk.jmh:jmh-generator-annprocess:1.28'
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle:61]: !!!Dependencies to JMH shouldn't fall into compile dependency. It should be used only for jmh тест

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

**Review by Nikita-Smirnov-Exactpro — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+# th2 Sailfish Utils (3.3.4)
 
 This library contains classes to convert messages from th2 to Sailfish format and vice versa. 
 They are used in several th2 projects to reuse Sailfish features: message comparison, codec/connect implementations, etc.
 
 ## Release Notes
 
+### 3.3.4
+
++ Added converter JavaType to class in `ProtoToIMessageConverter` to improve performance
```
Comment (jean-michel-perli-sonsin) [inline README.md:10]:
```suggestion
+ Added converter JavaType to the class in `ProtoToIMessageConverter` to improve performance
```

File: README.md [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
+
 ### 3.3.3
 
 + Add message properties to the Sailfish IMessage from th2 proto Message
```
Comment (jean-michel-perli-sonsin) [inline README.md:14]:
```suggestion
+ Added message properties to the Sailfish IMessage from th2 proto Message
```

File: src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java [hunk 4]
```diff
@@ -246,26 +261,49 @@ private Object convertSimple(Value listEvent, IFieldStructure fieldStructure) {
...
+        KindCase kindCase = value.getKindCase();
+        if (kindCase == KindCase.NULL_VALUE || kindCase == KindCase.KIND_NOT_SET) {
+            return null; // skip null value conversion
+        }
+        checkKind(value, fieldStructure.getName(), KindCase.SIMPLE_VALUE);
+        String simpleValue = value.getSimpleValue();
+        if (fieldStructure.isEnum()) {
+            simpleValue = convertEnumValue(fieldStructure, simpleValue);
+        }
+        // TODO may be place its logic into the MultiConverter
```
Comment (jean-michel-perli-sonsin) [inline src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java:273]:
```suggestion
        // TODO may place its logic into the MultiConverter
```

File: src/test/java/com/exactpro/th2/sailfish/utils/SampleBenchmark.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
```
Comment (jean-michel-perli-sonsin) [inline src/test/java/com/exactpro/th2/sailfish/utils/SampleBenchmark.java:2]:
```suggestion
 * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
```

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]
Hello, please see if you agree with the small changes proposed below:
- README.md
line 22/26 - **updated** Cradle version. **Introduced** async API for storing events

- src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java
line 2 - Copyright 2020-**2021** Exactpro (Exactpro Systems Limited)

Thanks!

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Review by jean-michel-perli-sonsin — APPROVED** [review]

**Pull request merged**